hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -253,10 +253,8 @@ getNormalizedFilePath (PluginId plId) uri = handleMaybe errMsg
         errMsg = T.unpack $ "Error(" <> plId <> "): converting " <> getUri uri <> " to NormalizedFilePath"
 
 -- ---------------------------------------------------------------------
-throwPluginError :: Monad m => PluginId -> String -> String -> ExceptT String m b
-throwPluginError (PluginId who) what where' = throwE msg
-    where
-        msg = (T.unpack who) <> " failed with " <> what <> " at " <> where'
+throwPluginError :: Monad m => String -> ExceptT String m b
```

Comment:
inline?

Reply:
I plan on coming back to this (assuming people are ok with my suggestion). I want to come up with some way to tie back log messages to their context. So for instance I'm thinking this would eventually return `ExceptT Context m b` where `Context` can provide some information for what happened. This `Context` should be usable in all locations where we log (so internal to `ghcide` or `shake` etc.) This way plugins can just give an error message and the plugin id is automatically attached.

Reply:
I'm not sure It would be in a follow up PR and the main detractor is whatever the solution is, has to cross `ghcide` and `hls-plugin-api` package barrier -- which I'm not sure how to deal with